shark/src/main/java/shark/HeapAnalyzer.kt

```diff
@@ -363,8 +363,11 @@ class HeapAnalyzer constructor(
       val className =
         recordClassName(graph.findObjectById(pathNode.objectId))
 
-      val firstLibraryLeakNode = if (rootNode is LibraryLeakNode) rootNode else
+      val firstLibraryLeakNode = if (rootNode is LibraryLeakNode) {
```

Comment:
I don't know that this is necessary for if statements returning a value.

Reply:
Detekt marks these as violations too. In `KeyedWeakReferenceMirror#39` we use nested if statements to return a value, so it's possible for statements to grow beyond simple `if-else`. Making it explicit will add extra safety.

Reply:
It's harder to make mistakes when returning values though? idk, I've grown used to single line `val foo = if (a) "a" else "b"`

Reply:
Single-line is indeed hard to make a mistake, that's why it is allowed to be without braces.
Multi-line is more opinionated. Having braces makes it easier to expand code later, which can be a case even for the return statements: in `KeyedWeakReferenceMirror#39` there's a variable being defined inside the `if` block, and it's quite possible that future modifications to the code will require extracting more variables. In these cases less lines of code will be considered modified, and one shouldn't worry about adding braces to a proper `else` branch, etc.
On the other hand, having explicit braces is overkill for most of the cases where simple `if-else` return value is used. 

Bottom line, Kotlin coding convention calls for always using braces for multi-line; this rules is also used by Detekt. We can suppress the rule on certain classes/methods, but can't configure it to ignore the return statements.